gateway-ha/src/main/java/com/lyft/data/gateway/ha/router/ResourceGroupsManager.java

```diff
@@ -1,42 +1,51 @@
 package com.lyft.data.gateway.ha.router;
 
+import com.sun.istack.Nullable;
 import java.util.List;
-
 import lombok.Data;
 import lombok.NonNull;
 import lombok.RequiredArgsConstructor;
 import lombok.ToString;
 
+
+
 public interface ResourceGroupsManager {
-  ResourceGroupsDetail createResourceGroup(ResourceGroupsDetail resourceGroup);
+  ResourceGroupsDetail createResourceGroup(ResourceGroupsDetail resourceGroup,
+                                           @Nullable String routingGroupDatabase);
 
-  List<ResourceGroupsDetail> readAllResourceGroups();
+  List<ResourceGroupsDetail> readAllResourceGroups(@Nullable String routingGroupDatabase);
```

Comment:
how about we use https://docs.oracle.com/javase/8/docs/api/java/util/Optional.html instead ?

Reply:
Dont get the advantage of using Optional instead of nullable- from what I read, Optional helps in deferring the logic to check null to the calling func because you dont have a null value, you just have an empty object. The conditional check for Nullable and Optional is not different (!= null and Optional.isPresent()) and is the equal amount of overhead on line https://github.com/lyft/presto-gateway/blob/31391d7d73a446b2c1dc31934a2b53622e42cb9f/gateway-ha/src/main/java/com/lyft/data/gateway/ha/persistence/JdbcConnectionManager.java#L30
any reason why you think that Optional would be better here?